p2p/net/reuseport/dialer.go

```diff
@@ -0,0 +1,91 @@
+package reuseport
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+
+	"github.com/libp2p/go-netroute"
+)
+
+type dialer struct {
+	specific    []*net.TCPAddr
+	loopback    []*net.TCPAddr
+	unspecified []*net.TCPAddr
```

Comment:
Can you add a comment here what kind of addresses would land in `specific` and in `unspecified`?